### ISSUE TITLE
Possibility to specify several rules for ttl.

### ISSUE THREAD

**Issue by R-omk** [metrico/gigapipe#158]
Sometimes different types of logs require different retention periods. (Or eviction to cold storage). 
To do this, it need the ability to distribute data across partitions over some subset of labels.

For example, I want to delete logs with a debug level in a couple of days, and keep all the rest much longer. 
Or another example, I would like to move logs with some set of labels that can provide analytical value to cold storage volume (S3 drive type ).

related  issue https://github.com/lmangani/cLoki/issues/75

**Comment by lmangani** [comment]
Good point, thanks for raising this! I guess we could implement some TTL-by-tag mechanism, but it might potentially ingestion down so it needs some discussion. cloki should work fine using the s3/minio storage engine, have you tried that already by any chance?
